rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -536,6 +553,90 @@ void saveOldFederation_after_RSKIP123_should_save_null() {
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideFederationBtcUTXOsTestArguments")
+    void getNewFederationBtcUTXOs(FederationStorageIndexKey federationStorageIndexKey, NetworkParameters networkParameters, ActivationConfig.ForBlock activations) {
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        Address btcAddress = BitcoinTestUtils.createP2PKHAddress(networkParameters, "test");
+
+        // Save utxos directly in storage
+        List<UTXO> expectedUtxos = BitcoinTestUtils.createUTXOs(2, btcAddress);
+        storageAccessor.saveToRepository(federationStorageIndexKey.getKey(), BridgeSerializationUtils.serializeUTXOList(expectedUtxos));
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Getting utxos from method
+        List<UTXO> actualUtxos = federationStorageProvider.getNewFederationBtcUTXOs(networkParameters, activations);
+
+        // Should be as the expected utxos
+        assertEquals(expectedUtxos, actualUtxos);
+
+    }
+
+    @Test
+    void getNewFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
+
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.only(ConsensusRule.RSKIP123).forBlock(0);
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        Address btcAddress = BitcoinTestUtils.createP2PKHAddress(networkParameters, "test");
+
+        DataWord newFederationBtcUtxosKey = NEW_FEDERATION_BTC_UTXOS_KEY.getKey();
+
+        // Save utxos directly in storage.
+        List<UTXO> expectedUtxos = BitcoinTestUtils.createUTXOs(1, btcAddress);
+        storageAccessor.saveToRepository(newFederationBtcUtxosKey, expectedUtxos, BridgeSerializationUtils::serializeUTXOList);
+
+        // Get utxos from method and check they are as expected
+        List<UTXO> actualUtxos = federationStorageProvider.getNewFederationBtcUTXOs(networkParameters, activations);
+        assertEquals(1, actualUtxos.size());
+        assertEquals(expectedUtxos, actualUtxos);
+
+        // Save new utxos directly in storage
+        List<UTXO> extraUtxos = new ArrayList<>(BitcoinTestUtils.createUTXOs(2, btcAddress));
+        storageAccessor.saveToRepository(newFederationBtcUtxosKey, extraUtxos, BridgeSerializationUtils::serializeUTXOList);
+
+        // Get utxos from method and check they are still the same as the original expected utxos since it is returning the cached utxos
+        List<UTXO> actualUtxosAfterSecondGet = federationStorageProvider.getNewFederationBtcUTXOs(null, activations);
+        assertEquals(1, actualUtxosAfterSecondGet.size());
+        assertEquals(expectedUtxos, actualUtxosAfterSecondGet);
+
+        // Get utxos directly from storage and confirm that the storage has the new utxos that the method didn't return because the method returned a cached list
```

Comment:
nit: I find this comment a bit long. I think the "that the method didn't return because the method returned a cached list" part is redundant, considering the comment above

Reply:
already merged lol